### PR TITLE
[extractor/amazon] Add `AmazonReviewsIE`

### LINKED ISSUE #5766: [Amazon] Can't download a product review video
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

I can't download a product review product from Amazon

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
yt-dlp -vU https://www.amazon.com/gp/customer-reviews/R10VE9VUSY19L3/ref=cm_cr_arp_d_rvw_ttl?ie=UTF8&ASIN=B09G9FPHY6
[1] 4014
[user@artix ~]$ [debug] Command-line config: ['-vU', 'https://www.amazon.com/gp/customer-reviews/R10VE9VUSY19L3/ref=cm_cr_arp_d_rvw_ttl?ie=UTF8']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.10.8 (CPython x86_64 64bit) - Linux-6.0.7-artix1-1-x86_64-with-glibc2.36.9000 (OpenSSL 3.0.7 1 Nov 2022, glibc 2.36.9000)
[debug] exe versions: ffmpeg N-109374-gc2a870222d (fdk,setts), ffprobe N-109374-gc2a870222d, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://www.amazon.com/gp/customer-reviews/R10VE9VUSY19L3/ref=cm_cr_arp_d_rvw_ttl?ie=UTF8
[generic] ref=cm_cr_arp_d_rvw_ttl?ie=UTF8: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] ref=cm_cr_arp_d_rvw_ttl?ie=UTF8: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.amazon.com/gp/customer-reviews/R10VE9VUSY19L3/ref=cm_cr_arp_d_rvw_ttl?ie=UTF8
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1485, in wrapper
    return func(self, *args, **kwargs)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1561, in __extract_info
    ie_result = ie.extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 674, in extract
    ie_result = self._real_extract(url)
  File "/home/user/.local/lib/python3.10/site-packages/yt_dlp/extractor/generic.py", line 2721, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.amazon.com/gp/customer-reviews/R10VE9VUSY19L3/ref=cm_cr_arp_d_rvw_ttl?ie=UTF8


[1]+  Exit 1                  yt-dlp -vU https://www.amazon.com/gp/customer-reviews/R10VE9VUSY19L3/ref=cm_cr_arp_d_rvw_ttl?ie=UTF8
```

### PR DESCRIPTION
**Pull request by bashonly**
Closes #5766
Addresses https://github.com/yt-dlp/yt-dlp/pull/5111#issuecomment-1363567678

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by dlcvietnam** [comment]
My test:

`
PS E:\Code\yt-dlp> yt-dlp https://www.amazon.com/Fitbit-Management-Intensity-Tracking-Included/dp/B0B5FGP237/
[AmazonStore] B0B5FGP237: Downloading webpage
[AmazonStore] B0B5FGP237: Downloading webpage
[AmazonStore] B0B5FGP237: Downloading webpage
WARNING: [AmazonStore] Invalid \escape in 'iew + what\\\'s new","': line 1 column 27203 (char 27202). Retrying (3/3)...

`

**Comment by bashonly** [comment]
@dlcvietnam are you sure you're running the PR branch? it works for me

```
$ yt-dlp -v --ignore-config 'https://www.amazon.com/Fitbit-Management-Intensity-Tracking-Included/dp/B0B5FGP237/'
[debug] Command-line config: ['-v', '--ignore-config', 'https://www.amazon.com/Fitbit-Management-Intensity-Tracking-Included/dp/B0B5FGP237/']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b644025b]
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python 3.10.8 (CPython x86_64 64bit) - Linux-6.0.12-arch1-1-x86_64-with-glibc2.36 (OpenSSL 1.1.1q  5 Jul 2022, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.2 (setts), ffprobe 5.1.2
[debug] Optional libraries: Cryptodome-3.12.0, brotlicffi-1.0.9.2, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1737 extractors
[AmazonStore] Extracting URL: https://www.amazon.com/Fitbit-Management-Intensity-Tracking-Included/dp/B0B5FGP237/
[AmazonStore] B0B5FGP237: Downloading webpage
WARNING: [AmazonStore] Unable to extract data. Retrying (1/3)...
[AmazonStore] B0B5FGP237: Downloading webpage
[download] Downloading playlist: Fitbit Inspire 3 Health &amp; Fitness Tracker with Stress Management, Workout Intensity, Sleep Tracking, 24\/7 Heart Rate and more, Morning Glow\/Black, One Size (S &amp; L Bands Included)
[AmazonStore] Playlist Fitbit Inspire 3 Health &amp; Fitness Tracker with Stress Management, Workout Intensity, Sleep Tracking, 24\/7 Heart Rate and more, Morning Glow\/Black, One Size (S &amp; L Bands Included): Downloading 6 items of 6
[download] Downloading item 1 of 6
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] ATVPDKIKX0DER: Downloading 1 format(s): 0
[debug] Invoking http downloader on "https://m.media-amazon.com/images/S/vse-vms-transcoding-artifact-us-east-1-prod/d2ac3038-42d3-4897-b8d5-65ac958f3362/default.jobtemplate.mp4.480.mp4"
[download] Destination: Fitbit Inspire 3 Product Overview [ATVPDKIKX0DER].mp4
[download]   7.7% of   25.99MiB at    6.41MiB/s ETA 00:03^C
ERROR: Interrupted by user
```

**Comment by dlcvietnam** [comment]
> 

yes. This is my debug

```
PS E:\Code\yt-dlp> yt-dlp -v --ignore-config 'https://www.amazon.com/Fitbit-Management-Intensity-Tracking-Included/dp/B0B5FGP237/'
[debug] Command-line config: ['-v', '--ignore-config', 'https://www.amazon.com/Fitbit-Management-Intensity-Tracking-Included/dp/B0B5FGP237/']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8 
[debug] yt-dlp version 2022.11.11 [8b644025b] 
[debug] Lazy loading extractors is disabled
[debug] Python 3.10.6 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1n  15 Mar 2022) 
[debug] exe versions: ffmpeg 4.3.2-2021-02-27-full_build-www.gyan.dev, ffprobe 4.3.2-2021-02-27-full_build-www.gyan.dev 
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, sqlite3-2.6.0, websockets-10.4 
[debug] Proxy map: {}
[debug] Loaded 1734 extractors 
[debug] [AmazonStore] Extracting URL: https://www.amazon.com/Fitbit-Management-Intensity-Tracking-Included/dp/B0B5FGP237/ 
[AmazonStore] B0B5FGP237: Downloading webpage 
WARNING: [AmazonStore] Invalid \escape in 'iew + what\\\'s new","': line 1 column 27203 (char 27202). Retrying (1/3)... 
[AmazonStore] B0B5FGP237: Downloading webpage 
WARNING: [AmazonStore] Invalid \escape in 'iew + what\\\'s new","': line 1 column 27203 (char 27202). Retrying (2/3)... 
[AmazonStore] B0B5FGP237: Downloading webpage 
WARNING: [AmazonStore] Invalid \escape in 'iew + what\\\'s new","': line 1 column 27203 (char 27202). Retrying (3/3)... 
[AmazonStore] B0B5FGP237: Downloading webpage 
ERROR: [AmazonStore] B0B5FGP237: Unable to extract data - Failed to parse JSON (caused by JSONDecodeError('Invalid \\escape in \'iew + what\\\\\\\'s new","\': line 1 column 27203 (char 27202)')); please report this issue on  https:/
/github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

File: yt_dlp/extractor/amazon.py [hunk 1]
```diff
@@ -20,28 +32,32 @@ class AmazonStoreIE(InfoExtractor):
...
             },
-        }]
+        }],
+        'expected_warnings': ['Unable to extract data'],
     }, {
         'url': 'https://www.amazon.in/Sony-WH-1000XM4-Cancelling-Headphones-Bluetooth/dp/B0863TXGM3',
         'info_dict': {
             'id': 'B0863TXGM3',
-            'title': 'md5:d1d3352428f8f015706c84b31e132169',
+            'title': 'md5:f62b7bc5591a6107553bfff00d6e49fc',
```
Comment (pukkandan) [inline yt_dlp/extractor/amazon.py:41]: If these are not static, use weaker test definition. Eg:
```suggestion
            'title': str,
```

**Review by pukkandan — APPROVED** [review]

**Pull request merged**